internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,378 @@
+			log.Trace("no token provided, will attempt to run authless trigger")
+			requireAuth = false
+		}
+
+		// Connect back to our own gRPC service.
+		grpcConn, err := serverclient.Connect(ctx,
+			serverclient.Logger(log),
+			serverclient.FromContextConfig(&clicontext.Config{
+				Server: serverconfig.Client{
+					Address:     addr,
+					RequireAuth: requireAuth,
+					AuthToken:   token,
+
+					// Our gRPC server should always be listening on TLS.
+					// We ignore it because its coming out of our own process.
+					Tls:           tls,
+					TlsSkipVerify: true,
+				},
+			}),
+		)
+		if err != nil {
+			log.Error("trigger connection back to gRPC failed", "err", err)
+			return
+		}
+		defer grpcConn.Close()
+
+		// Our API client
+		client := pb.NewWaypointClient(grpcConn)
+
+		requestVars := mux.Vars(r)
+		runTriggerId := requestVars["id"]
+
+		overrideVarJSONRaw := r.URL.Query().Get("variables")
+		var (
+			vo                map[string]string
+			variableOverrides []*pb.Variable
+		)
+
+		if overrideVarJSONRaw != "" {
+			if err := json.Unmarshal([]byte(overrideVarJSONRaw), &vo); err != nil {
+				http.Error(w, fmt.Sprintf("failed to decode 'variables' json request param into a map: %s", err), 500)
+				return
+			}
+
+			for name, value := range vo {
+				v := &pb.Variable{
+					Name:   name,
+					Source: &pb.Variable_Cli{Cli: &empty.Empty{}},
+				}
+
+				if valBool, err := strconv.ParseBool(value); err == nil {
+					v.Value = &pb.Variable_Bool{Bool: valBool}
+				} else if valInt, err := strconv.ParseInt(value, 10, 64); err == nil {
+					v.Value = &pb.Variable_Num{Num: valInt}
+				} else {
+					// NOTE: for this case, it can either be a "string" or
+					// complex HCL type like an array or map. We can set this value
+					// as a Variable_Str here, and when we go to parse the variables
+					// later we do the proper string versus HCL check in variables.go
+					v.Value = &pb.Variable_Str{Str: value}
+				}
+
+				variableOverrides = append(variableOverrides, v)
+			}
+		}
+
+		var resp *pb.RunTriggerResponse
+		runTriggerReq := &pb.RunTriggerRequest{
+			Ref: &pb.Ref_Trigger{
+				Id: runTriggerId,
+			},
+			VariableOverrides: variableOverrides,
+		}
+
+		// attempt to make a grpc request to run trigger by id
+		if requireAuth {
+			resp, err = client.RunTrigger(ctx, runTriggerReq)
+		} else {
+			resp, err = client.NoAuthRunTrigger(ctx, runTriggerReq)
+		}
+		if err != nil {
+			log.Error("server failed to run trigger", "id", runTriggerId, "err", err)
+
+			if status.Code(err) == codes.PermissionDenied {
+				http.Error(w, fmt.Sprintf("request not authorized to run trigger: %s", err), 401)
+			} else {
+				// improve http error code, which is more applicable for general queue failures?
+				http.Error(w, fmt.Sprintf("server failed to run trigger: %s", err), 412)
+			}
+
+			return
+		}
+		if resp == nil {
+			http.Error(w, fmt.Sprintf("server returned no job ids from run trigger %q", runTriggerId), 500)
```

Comment:
`RunTrigger()` (and I assume `NoAuthRunTrigger()`) [can return a gRPC error code of `NotFound`](https://github.com/hashicorp/waypoint/blob/133491b031efb585a34151ed26b29958d952b27f/internal/server/singleprocess/state/db.go#L107), which I don't think should be considered a `500` internal server error. We should either check for that not found error and pass it on, or otherwise just return a `404` here either way

Reply:
Oh, I just checked out the code here...this is happening right above here!

Reply:
👍